### PR TITLE
parsers: add ccp-guinea

### PR DESCRIPTION
**Pull request by abhidg**
*Notes*: Source file has encoding iso-8859-1, run with adtl option `--encoding`

*Questions*:

- [ ] How to map treatments before hospitalisation?
- [ ] Gestational assessment weeks has 2 entries with 'SA', what does that mean?

*Validation*:

|table       	|valid	|total	|percentage_valid|
|---------------|-------|-------|----------------|
|subject       	|2720	|2741	|99.233856% |
|visit         	|2665	|2741	|97.227289% |
|observation   	|203231	|219401	|92.629933% |

## subject

* 19: data must contain ['admission_date', 'age', 'country_iso3', 'pathogen', 'sex_at_birth', 'subject_id'] 
properties
* 2: data.pregnancy_gestational_assessment_weeks must be number

## visit

* 76: data must contain ['visit_id', 'subject_id', 'country_iso3', 'start_date', 'outcome', 'date_outcome'] 
properties

## observation

* 16170: data must contain ['phase', 'date', 'name'] properties

### CODE REVIEW

File: isaric/parsers/ccp-guinea.toml [hunk 1]
```diff
@@ -0,0 +1,1119 @@
...
+  name = "other_symptom"
+  phase = "study"
+  date = { field = "other_2_v2_date" }
+  text = { field = "other_2_v2", description = "Other symptom 3" }
+
+[[observation]]
+  name = "other_symptom"
+  phase = "study"
+  date = { field = "treatment_date_t" }
+  text = { field = "premeds_other_meds2", description = "Other treatment (3)" }
```
Comment (ekamau) [inline isaric/parsers/ccp-guinea.toml:1039]: questionable ...?

**Review by ekamau — APPROVED** [review]
looks good ...

**Comment by sadiekelly** [comment]
Gestational age with SA: we took this to mean weeks (semaine in French) so just used to number given as the gestational age. One patient had gestational age of 4SA, which seemed unlikely. Upon review they were postpartum and this represented weeks postpartum rather than gestational age

**Comment by sadiekelly** [comment]
mapping medications pre-admission; my thoughts when reviewing RAPID data yesterday were to put pre-admission and chronic medications in the subject table, along with comorbidities. An issue though is that we have vaccination data relating to vaccination status pre-admission and also vaccination data from follow-up, indicating vaccinations received both pre-admission and since admission, with considerable overlap and inconsistencies. The vaccination data relating to pre-admission could also go in subject, but we'd also then have vaccination data from follow up in observations, some data will match the subject table, some will be additional data for further vaccinations received, and some will not match at all due to many reasons (data entry, patient recall etc.). Something to discuss further?

File: isaric/parsers/ccp-guinea.toml [hunk 2]
```diff
@@ -0,0 +1,1119 @@
...
+  field = "overall_invasive_proccur"
+  description = "Invasive ventilation?"
+  ref = "Y/N/NK"
+
+[visit.treatment_noninvasive_ventilation]
+  field = "overall_noninvasive_proccur"
+  description = "Non-invasive ventilation (eg BiPAP, CPAP)?"
+  ref = "Y/N/NK"
+
+[visit.treatment_other]
```
[the remark this answers is not in the document]
Comment (abhidg) [reply isaric/parsers/ccp-guinea.toml:577]: Good point, changed to fields which are during visit

**Comment by abhidg** [comment]
@sadiekelly currently we are not encoding vaccinations and meds pre-admission, let's come up with a plan for this

**Pull request merged**